### DISCUSSION TITLE
vitest html reports as static webpage (no server)??

### DISCUSSION THREAD

**Discussion by pswenson** [vitest-dev/vitest#6927] [Q&A] [answered]
The vitest html test reports are great for local dev, however we'd like to see the reports in CICD.  Can this report be served up as a static web page?  If I remove the cors/file restrictions on my browser the HTML renders but doesn't appear to actually load the test information (I get a JSON parse error).  

Or will I have to spin up a web server and copy the assets there to be able to view as desired?  Thanks!

**Comment by hi-ogawa** [comment, marked as answer]
> Can this report be served up as a static web page? 

That should be possible. After running with html reporter https://vitest.dev/guide/reporters.html#html-reporter, it's expected that output directory `html` can be served as spa.

**Reply by pswenson** [reply]
Yes, this is exactly what I tried however it serves as an HTML page with no test data loaded.  Same result in both Safari and Chrome

it works fine with `npx vite preview --outDir html`

**Reply by hi-ogawa** [reply]
There might be an issue with gzip decoding. Can you show what response headers you see for `/html.meta.json.gz` in devtools?

**Reply by hi-ogawa** [reply]
On a related note, there's actually something broken on `vite preview` (the actual cause is in `sirv` https://github.com/lukeed/sirv/pull/161) which is worked around on Vitest side https://github.com/vitest-dev/vitest/pull/3113#issuecomment-1492403214
Probably your gz asset needs to be served explicitly with `content-type: application/gzip` response header.

**Reply by pswenson** [reply]
moving the assets to an S3 bucket worked fine.. which is really what we are after anyway.  thx for the tips!

**Reply by hi-ogawa** [reply]
This might be something we can improve https://github.com/vitest-dev/vitest/pull/6928, so I still want to know more details about your previous setup where `html.meta.json.gz` was broken. Was it a common CI/CD solution which other people might come across?

**Reply by Uttu316** [reply]
@hi-ogawa can we generate and use JSON file instead of a gzip file while building the report?

**Reply by hi-ogawa** [reply]
@Uttu316 No, currently they are hard-coded to be gzip since https://github.com/vitest-dev/vitest/pull/3113.
